src/App.tsx

```diff
@@ -1,24 +1,228 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Notification } from './components/Notification';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { ToodList } from './components/TodoList/TodoList';
+import {
+  getTodos,
+  postTodo,
+  deleteTodo, patchTodo,
+} from './api/todos';
+import { Loader } from './components/Loader';
+import { FilterType } from './types/FilterEnum';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const USER_ID = 7006;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(FilterType.All);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        setErrorMessage('');
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        if (error instanceof Error) {
+          setErrorMessage(error.message);
+        }
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const addTodo = (todoTitle: string) => {
+    const newTodo = {
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setIsInputDisabled(true);
+
+    setTemporaryTodo({ ...newTodo, id: 0 });
+
+    postTodo(newTodo)
+      .then(result => {
+        setTodos(state => [...state, result]);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setIsInputDisabled(false);
+        setTemporaryTodo(undefined);
```

Comment:
Let's use one approach, use either promise chaining methods or async/await syntax (async/await is more preferable)